### ISSUE TITLE
Reservations can be cancelled

### ISSUE THREAD

**Issue by developeregrem** [developeregrem/pve#15]
It must be possible to cancel a reservation.
A new button should be added to the modal (click on Reservation).
When you click on the button, the reservation should "disappear" and be marked as cancelled in the DB. (The room can be reassigned).
A new attribute must be inserted in the Reservation entity to indicate whether a reservation has been cancelled or not.
Only reservations that have not been canceled can be displayed in the reservation overview.
At the same time, the check to see whether a room is available (create / edit new reservation) must also be adjusted so that only reservations that have not been canceled are queried.

Canceled reservations are to be evaluated later for the statistics module.

**Comment by daking88** [comment]
what about doing this via the "status" of a reservation? adding a new status is easy, we then just need to think about a "fixed" canceled status. we could then as well easily filter for the records.

**Comment by developeregrem** [comment]
Yes would be an option. Some additional thoughts that came into my mind:
- should canceled reservations still be visible/accessible from the UI or should we just take this for statistical analysis (just a flag in the database)
- if only as "ghost" in the database the change to "canceled" can not be undone a new reservation must be created if this was a mistake
- if still be visible in the UI -> HOW? Currently, only one reservation can be displayed for the same apartment and period. If we have one canceled and afterwards one new confirmed reservation, this is not possible atm. However, it could be possible to resolve this conflict with #70 where I'm currently working on.

**Comment by daking88** [comment]
To be honest, I wouldn't show it in the calendar and rather hide it. It would only confuse. It does not necessarily mean we can't get it back, or? How about a sub menu viewing all cancelled and there you can revert the change. Would then more or less be a state change, as said above.

**Comment by developeregrem** [comment]
added a new sytem reservation status for cancelled / no-show reservations.
